Automattic/wp-calypso
# Signup: Add pure versions of steps/flows configs

While working on #25387, I discovered that `signup/config/flows` and `signup/config/steps` introduce a DOM API dependency due to imports such as `lib/signup/step-actions` and `lib/user`. 

In order to facilitate handling of the signup data flow via Redux, it is necessary to create pure, side-effects-free versions of these configuration files.

To this end, I have moved the flows and steps definitions into `signup/config/flows-pure` and `signup/config/steps-pure`, respectively. These pure counterparts export their configuration values without DOM API dependent values or functions. `signup/config/flows` and `signup/config/steps` have been kept backwards compatible for when such values or functions remain necessary.

In summary:
1. This PR creates `signup/config/flows-pure` and `signup/config/steps-pure` configuration files. These files replace their DOM dependent functions with noop's.
2. This PR preserves backward compatibliity for `signup/config/flows` and `signup/config/steps`.
3. Signup's dependency store, progress store, and flow controller have been refactored to use `signup/config/steps-pure`.

# Testing Instructions
1. Spin up this branch locally.
2. Go through the signup flow at [`/start`](http://calypso.localhost:3000/start/domains).
3. Ensure no behavioral changes have been introduced.
